### PR TITLE
[Anaconda] Update cryptography package due to GHSA-jfhm-5ghh-2f97

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Dev container name**: 
* Anaconda

**Description**: 

This PR patches the following vulnerability:

* [GHSA-jfhm-5ghh-2f97](https://github.com/advisories/GHSA-jfhm-5ghh-2f97) - related to the `cryptography` package;

This vulnerability come from the continuumio/anaconda3 image used upstream for the anaconda devcontainer.

*Changelog*:

* Updated Dockerfile
  * Locked version for patched Conda package;
    * `cryptography` - _minimum package version set to `41.0.7`_; 
  
* Updated tests to verify `cryptography` minimum version (Minimum package version set to `41.0.7` which fixes [GHSA-jfhm-5ghh-2f97](https://github.com/advisories/GHSA-jfhm-5ghh-2f97));

**Checklist**:

* [x]  Checked that applied changes work as expected

### CODE REVIEW

File: src/anaconda/test-project/test.sh [hunk 1]
```diff
@@ -40,16 +40,15 @@ checkPythonPackageVersion "nbconvert" "6.5.1"
 checkPythonPackageVersion "werkzeug" "2.2.3"
 checkPythonPackageVersion "certifi" "2022.12.07"
 checkPythonPackageVersion "requests" "2.31.0"
-checkPythonPackageVersion "cryptography" "41.0.3"
-checkPythonPackageVersion "transformers" "4.30.0"
```
Comment (alexander-smolyakov) [inline src/anaconda/test-project/test.sh:44]: It looks like an unrelated change, please revert it.

File: src/anaconda/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -11,7 +11,7 @@ RUN conda install \
     # https://github.com/advisories/GHSA-j7hp-h8jx-5ppr
     pillow=10.0.1 \
     # https://github.com/advisories/GHSA-v845-jxx5-vc9f
-    urllib3==1.26.18
+    urllib3==1.26.18
```
Comment (alexander-smolyakov) [inline src/anaconda/.devcontainer/Dockerfile:14]: Nit: extra whitespace

File: src/anaconda/.devcontainer/Dockerfile [hunk 3]
```diff
@@ -31,7 +31,9 @@ RUN python3 -m pip install --upgrade \
     # https://github.com/advisories/GHSA-r726-vmfq-j9j3 
     jupyter_server==2.7.2 \
     # https://github.com/advisories/GHSA-5wvp-7f3h-6wmm
-    pyarrow==14.0.1
+    pyarrow==14.0.1 \
+    # https://github.com/advisories/GHSA-jfhm-5ghh-2f97
+    cryptography==41.0.6
```
Comment (alexander-smolyakov) [inline src/anaconda/.devcontainer/Dockerfile:36]: As I can see, the Anaconda default channel contains `cryptography` v41.0.7: https://anaconda.org/anaconda/cryptography/files

Could you rework the patch to install the package from the conda channel? For compatibility sake.

**Review by alexander-smolyakov — CHANGES_REQUESTED** [review]

File: src/anaconda/test-project/test.sh (hunk 1 above)
Comment (gauravsaini04) [reply src/anaconda/test-project/test.sh:44]: Have corrected it. Thanks Alex.

File: src/anaconda/.devcontainer/Dockerfile (hunk 2 above)
Comment (gauravsaini04) [reply src/anaconda/.devcontainer/Dockerfile]: Have corrected it

File: src/anaconda/.devcontainer/Dockerfile (hunk 3 above)
Comment (gauravsaini04) [reply src/anaconda/.devcontainer/Dockerfile]: Corrected it.

**Review by alexander-smolyakov — APPROVED** [review]

**Review by eljog — APPROVED** [review]

**Comment by gauravsaini04** [comment]
Hi @eljog, have resolved merge conflicts

**Review by eljog — APPROVED** [review]

**Pull request merged**